crates/topos-p2p/src/runtime/mod.rs

```diff
@@ -95,4 +146,29 @@ impl Runtime {
 
         Ok(())
     }
+
+    pub(crate) fn healthy_status_changed(&mut self) -> Option<Event> {
+        let behaviours = self.swarm.behaviour();
+        let gossipsub = &behaviours.gossipsub.health_status;
+        let discovery = &behaviours.discovery.health_status;
+
+        let new_status = match (discovery, gossipsub) {
+            (HealthStatus::Initializing, _) | (_, HealthStatus::Initializing) => {
+                HealthStatus::Initializing
+            }
+            (HealthStatus::Unhealthy, _) | (_, HealthStatus::Unhealthy) => HealthStatus::Unhealthy,
+            (HealthStatus::Recovering, _) | (_, HealthStatus::Recovering) => {
+                HealthStatus::Recovering
+            }
+            (HealthStatus::Healthy, HealthStatus::Healthy) => HealthStatus::Healthy,
+        };
```

Comment:
If we have one `initializing` and the other `unhealthy` that will be the first match so we'll consider being `initializing`? If so maybe we wanna reflect the "unhealthyness" as top priority by having it as 1st branch?

Reply:
Yeah, on the `behaviour` I was thinking that we could wait for every behaviour to finish the `initialization` phase before deciding if the `layer` is `unhealthy`. But we may need another variant such as `Killing` defining that the node need to kill itself.